scripts/globals/battlefield.lua

```diff
@@ -68,10 +70,788 @@ xi.battlefield.dropChance =
+
+                        if current > roll then
+                            if entry.itemid == 0 then
+                                break
+                            end
+
+                            if entry.itemid == 65535 then
+                                local gil = entry.amount/#players
+
+                                for k = 1, #players, 1 do
+                                    players[k]:addGil(gil)
+                                    players[k]:messageSpecial(zones[players[1]:getZoneID()].text.GIL_OBTAINED, gil)
+                                end
+
+                                break
+                            end
+
+                            players[1]:addTreasure(entry.itemid, npc)
+                            break
+                        end
+                    end
+                end
+            end
+        end
+    end
+end
+
+function xi.battlefield.getBattlefieldOptions(player, npc, trade)
+    local result = 0
+    local contents = xi.battlefield.contentsByZone[player:getZoneID()]
+
+    if contents == nil then
+        return result
+    end
+
+    for _, content in ipairs(contents) do
+        if content:checkRequirements(player, npc, true, trade) and not player:battlefieldAtCapacity(content.battlefieldId) then
+            result = utils.mask.setBit(result, content.index, true)
+        end
+    end
+
+    return result
+end
+
+function xi.battlefield.rejectLevelSyncedParty(player, npc)
+    for _, member in pairs(player:getAlliance()) do
+        if member:isLevelSync() then
+            local zoneId = player:getZoneID()
+            local ID = zones[zoneId]
+            -- Your party is unable to participate because certain members' levels are restricted
+            player:messageText(npc, ID.text.MEMBERS_LEVELS_ARE_RESTRICTED, false)
+            return true
+        end
+    end
+    return false
+end
+
+BattlefieldMission = setmetatable({ }, { __index = Battlefield })
+BattlefieldMission.__index = BattlefieldMission
+BattlefieldMission.__eq = function(m1, m2)
+    return m1.name == m2.name
+end
+
+BattlefieldMission.isMission = true
+
+-- Creates a new Limbus Battlefield interaction
+-- Data takes the additional following keys:
+--  - missionArea: The mission area this battlefield is associated with (optional)
+--  - mission: The mission this battlefield is associated with (optional)
+--  - missionStatusArea: The mission area to retrieve the mission status from. Will default to using the player's nation (optional)
+--  - missionStatus: The optional extra status information xi.mission.status (optional)
+--  - requiredMissionStatus: The required mission status to enter
+--  - skipMissionStatus: The required mission status to skip the cutscene. Defaults to the required mission status.
+--  - canLoseExp: Determines if a character loses experience points upon death while inside the battlefield. Defaults to false. (optional)
+function BattlefieldMission:new(data)
```

Comment:
Similar to a previous example.  There's duplicated comments above and below; is there a way we can consolidate this info?

Reply:
I'll address this once we figure out the primary one